src/internal/m365/exchange/contacts_restore.go

```diff
@@ -94,22 +99,40 @@ func restoreContact(
 	}
 
 	ctx = clues.Add(ctx, "item_id", ptr.Val(contact.GetId()))
-	collisionKey := api.ContactCollisionKey(contact)
 
-	if _, ok := collisionKeyToItemID[collisionKey]; ok {
+	var (
+		collisionKey         = api.ContactCollisionKey(contact)
+		collisionID          string
+		shouldDeleteOriginal bool
+	)
+
+	if id, ok := collisionKeyToItemID[collisionKey]; ok {
 		log := logger.Ctx(ctx).With("collision_key", clues.Hide(collisionKey))
 		log.Debug("item collision")
 
-		// TODO(rkeepers): Replace probably shouldn't no-op.  Just a starting point.
-		if collisionPolicy == control.Skip || collisionPolicy == control.Replace {
+		if collisionPolicy == control.Skip {
 			log.Debug("skipping item with collision")
 			return nil, graph.ErrItemAlreadyExistsConflict
 		}
+
+		collisionID = id
```

Comment:
looking at this now, I'm wondering if it's a little simplistic. It feels like we may run afoul of the case where we get some 5xx error but the item is actually created. The run will fail (5xx for post) but now we'll potentially have 2 items with the same collision key: the original original and the one that failed with the 5xx

Reply:
Yeah, fair point.  We'll have to handle multiple-item collisions gracefully.  After all, for certain category types, that condition can be fairly easily manufactured using the api or webapp.  It'll only turn into a concern when restoring with replace behavior; and in that case we're only going to replace one of the many colliding items.  Hope is that fits around all but the most extreme issues without too much fuss.

Reply:
would just keeping a list of IDs in the collision cache (not sure if that's the right name) suffice?

Reply:
It'd be a way to identify all the associated items.  But if we're not going to use them for anything, I don't see much purpose in the aggregation.